include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -854,9 +858,17 @@ struct OrtApi {
   /**
      * \param value A tensor created from OrtCreateTensor... function.
      * \param s A null terminated string.
-     * \param index index of string tensor element to fill 
+     * \param index index of string tensor element to fill
      */
   ORT_API2_STATUS(FillStringTensorElement, _Inout_ OrtValue* value, _In_ const char* s, size_t index);
+
+  /**
+     * Set a single session configuration entry as a pair of an enum and a string
+     * \param config_key An enum as the key of the configuration
+     * \param config_value A null terminated string representation of the config value
+     */
+  ORT_API2_STATUS(SetSessionConfiguration, _Inout_ OrtSessionOptions* options,
+                  _In_ OrtSessionConfigKey config_key, _In_ const char* config_value);
```

Comment:
Should we allow users to set existing session options as well using this method?

Reply:
We'd have to implement logic to handle converting the string to the correct type including error checking for all options. I think it would also be confusing to have two different ways to set one value in SessionOptions in the API.

---
In reply to: [466694258](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466694258) [](ancestors = 466694258)

Reply:
I wanted to convert logid (which is a good candidate to use the map<enum, string>) into the the map, but it is used in too many places

---
In reply to: [466726344](https://github.com/microsoft/onnxruntime/pull/4718#discussion_r466726344) [](ancestors = 466726344,466694258)